etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/database/PostgresDao.java

```diff
@@ -45,7 +57,8 @@ public void upsertData(String tableName, List<DbColumn> values, String conflictC
 
         sqlStatementBuilder.append(") VALUES (");
 
-        sqlStatementBuilder.append("?, ".repeat(values.size()));
+        sqlStatementBuilder.append("?, ".repeat(values.size() - MESSAGE_HD_DB_COLUMNS.size()));
+        sqlStatementBuilder.append("?::JSONB, ".repeat(MESSAGE_HD_DB_COLUMNS.size()));
```

Comment:
The `upsertData` method is meant to be generic and completely divorced from metadata.  This change wouldn't allow for it to be generic.  Can we move the JSONB and JSON formatting to `DatabasePartnerMetadataStorage#saveMetadata`?

Reply:
Maybe we can meet up to discuss this further. I have some thoughts on this and would like your input.

Reply:
Hell yeah, definitely!